router/xgress/payload_buffer.go

```diff
@@ -32,15 +33,29 @@ type PayloadBufferForwarder interface {
 }
 
 type PayloadBufferController struct {
-	buffers   cmap.ConcurrentMap // map[sessionId.Token]*PayloadBuffer
+	buffers   cmap.ConcurrentMap // map[bufferId{address+sessionId}]*PayloadBuffer
+	sessions  cmap.ConcurrentMap // map[sessionId][]string{bufferId{address+sessionId}
 	forwarder PayloadBufferForwarder
 }
 
 func NewPayloadBufferController(forwarder PayloadBufferForwarder) *PayloadBufferController {
-	return &PayloadBufferController{
+	pbc := &PayloadBufferController{
 		buffers:   cmap.New(),
+		sessions:  cmap.New(),
 		forwarder: forwarder,
 	}
```

Comment:
Do we want to either remove this or add a config flag to control it?

Reply:
i'm still cleaning this up. wait until i add reviewers. :-)